src/components/editor/Store/types.ts

```diff
@@ -1,5 +1,15 @@
 import { AlertColor } from '@mui/material';
-import { DraftSpec } from 'hooks/useDraftSpecs';
+import { PostgrestError } from '@supabase/postgrest-js';
+import { SuccessResponse } from 'hooks/supabase-swr';
+import { DraftSpec, DraftSpecQuery } from 'hooks/useDraftSpecs';
+import { KeyedMutator } from 'swr';
+
+export interface DraftSpecResponse {
+    draftSpecs: DraftSpecQuery[];
+    isValidating: boolean;
+    mutate: KeyedMutator<SuccessResponse<DraftSpecQuery>> | null;
```

Comment:
Not sure on this one yet... but given how often we have to check for this function I am thinking we should not make this null and rather just default it to a noop instead.